### ISSUE TITLE
Can't build on Apple M1/ MacOS Sonoma

### ISSUE THREAD

**Issue by ddppddpp** [pycompression/python-zlib-ng#35]
Context:
New, empty virtual environment. Python 3.11.5.
```
python -m pip install zlib-ng
Collecting zlib-ng
  Downloading zlib-ng-0.4.1.tar.gz (2.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.5/2.5 MB 13.8 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: zlib-ng
  Building wheel for zlib-ng (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for zlib-ng (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [50 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-13.5-arm64-cpython-311
      creating build/lib.macosx-13.5-arm64-cpython-311/zlib_ng
      copying src/zlib_ng/__init__.py -> build/lib.macosx-13.5-arm64-cpython-311/zlib_ng
      copying src/zlib_ng/gzip_ng_threaded.py -> build/lib.macosx-13.5-arm64-cpython-311/zlib_ng
      copying src/zlib_ng/gzip_ng.py -> build/lib.macosx-13.5-arm64-cpython-311/zlib_ng
      copying src/zlib_ng/zlib_ng.pyi -> build/lib.macosx-13.5-arm64-cpython-311/zlib_ng
      copying src/zlib_ng/py.typed -> build/lib.macosx-13.5-arm64-cpython-311/zlib_ng
      creating build/lib.macosx-13.5-arm64-cpython-311/zlib_ng/zlib-ng
      copying src/zlib_ng/zlib-ng/LICENSE.md -> build/lib.macosx-13.5-arm64-cpython-311/zlib_ng/zlib-ng
      copying src/zlib_ng/zlib-ng/README.md -> build/lib.macosx-13.5-arm64-cpython-311/zlib_ng/zlib-ng
      running build_ext
      Checking for compiler... /Library/Developer/CommandLineTools/usr/bin/clang -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
      Checking for Symbol versioning... No.
      Checking for shared library support... Building shared library libz-ng.2.1.5.dylib with /Library/Developer/CommandLineTools/usr/bin/clang -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include.
      Checking for off64_t... No.
      Checking for _off64_t... No.
      Checking for fseeko... Yes.
      Checking for posix_memalign... Yes.
      Checking for aligned_alloc... Yes.
      Checking for strerror... Yes.
      Checking for getauxval() in sys/auxv.h... No.
      Checking for unistd.h... Yes.
      Checking for ptrdiff_t... Yes.
      Checking for ANSI C compliant compiler...  Yes.
      Checking for -fno-semantic-interposition... Yes.
      Checking for -fno-lto... Yes.
      Checking for attribute(visibility(hidden)) support... Yes.
      Checking for attribute(visibility(internal)) support... Yes.
      Checking for attribute(aligned) ... Yes.
      Checking for _Thread_local ... Yes.
      Checking for __builtin_ctz ... Yes.
      Checking for __builtin_ctzll ... Yes.
      ARM floating point arch: -mfloat-abi=softfp
      Checking for arm_acle.h... Yes.
      Check whether -march=armv8-a+crc works ... Yes.
      Check whether -march=armv8-a+simd is available ... Yes.
      check whether compiler supports 4 wide register loads ... Yes.
      Check whether -march=armv6 works ... No.
      Checking for ARMv6 inline assembly ... No.
      Checking for ARMv6 intrinsics ... No.
      Uname: Darwin
      ARCH: aarch64
      Using arch directory: arch/arm
      Architecture-specific static object files: arm_features.o crc32_acle.o insert_string_acle.o adler32_neon.o chunkset_neon.o compare256_neon.o slide_hash_neon.o
      Architecture-specific shared object files: arm_features.lo crc32_acle.lo insert_string_acle.lo adler32_neon.lo chunkset_neon.lo compare256_neon.lo slide_hash_neon.lo
      error: [Errno 2] No such file or directory: 'gmake'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for zlib-ng
Failed to build zlib-ng
ERROR: Could not build wheels for zlib-ng, which is required to install pyproject.toml-based projects
```

**Comment by rhpvorderman** [comment]
It is true that macos arm wheels are not provided as I can't build and test these yet. (Although I believe there are some github developments in this regard). 

So it will try to build on your computer. Given that the python binding code is nothing special in terms of build complexity, most of the issues will originate from building zlib-ng itself, which is the case here. My advice would be to check the documentation about building on the [zlib-ng homepage](https://github.com/zlib-ng/zlib-ng).

Also the error messages in this case are very constructive: "error: [Errno 2] No such file or directory: 'gmake'". Which indicates you are missing software that is required to build the project. In this case Gnu Make which is different from the included make on MacOS and thus uses the `gmake` command. 

Does this help you?

**Comment by rhpvorderman** [comment]
In the meantime I can work on getting wheels out there: https://github.com/orgs/community/discussions/102846

**Comment by ddppddpp** [comment]
Hello,
A simple clone of the zlib-ng and a classic configre/make/sudo make install completes correctly and produces no errors on the same system.

**Comment by rhpvorderman** [comment]
Ah. I see this is indeed a failure in the build-system of this library:

```python
    if sys.platform == "darwin":  # Cmake does not work properly
        subprocess.run([os.path.join(build_dir, "configure")], **run_args)
        subprocess.run(["gmake", "libz-ng.a"], **run_args)
``` 
I will try to fix this before the next release.

**Comment by rhpvorderman** [comment]
Sorry for my late reply @ddppddpp  does `pip install git+https://github.com/pycompression/python-zlib-ng.git@macosm1` work for you?

**Comment by ddppddpp** [comment]
I get an error for a missing requirements.txt file

**Comment by rhpvorderman** [comment]
That's weird. Python-zlib-ng has no requirements. Can you paste the error here?

**Comment by ddppddpp** [comment]
my 2nd attempt worked, which is strange, as I didn't do anything different:
```
 pip install git+https://github.com/pycompression/python-zlib-ng.git@macosm1
Collecting git+https://github.com/pycompression/python-zlib-ng.git@macosm1
  Cloning https://github.com/pycompression/python-zlib-ng.git (to revision macosm1) to /private/var/folders/08/vnp5z07n55b22_j0z72brvy40000gn/T/pip-req-build-stzepl5b
  Running command git clone --filter=blob:none --quiet https://github.com/pycompression/python-zlib-ng.git /private/var/folders/08/vnp5z07n55b22_j0z72brvy40000gn/T/pip-req-build-stzepl5b
  Running command git checkout -b macosm1 --track origin/macosm1
  Switched to a new branch 'macosm1'
  Branch 'macosm1' set up to track remote branch 'macosm1' from 'origin'.
  Resolved https://github.com/pycompression/python-zlib-ng.git to commit 39b8832983ad4ddfcfe909cf8de0fe4484ec10e8
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: zlib-ng
  Building wheel for zlib-ng (pyproject.toml) ... done
  Created wheel for zlib-ng: filename=zlib_ng-0.5.0.dev0-cp311-cp311-macosx_13_0_arm64.whl size=92208 sha256=a786eaddea3af6b52ab10578cea3a8c301865fd00184be3e1d8f729b2f3d541d
  Stored in directory: /private/var/folders/08/vnp5z07n55b22_j0z72brvy40000gn/T/pip-ephem-wheel-cache-gt08j5d4/wheels/6c/94/da/aaebcef50c13587461fae1716caaacebaae13ca6ed7fcc193e
Successfully built zlib-ng
Installing collected packages: zlib-ng
Successfully installed zlib-ng-0.5.0.dev0
```

**Comment by rhpvorderman** [comment]
Okay great. Then it seems as  if my patch solved your problem. I can go ahead and release it.

**Comment by rhpvorderman** [comment]
Thanks for reporting back.This was fixed as of 0.4.2 so I will close this issue.
